src/main/java/com/artipie/RepoConfig.java

```diff
@@ -86,6 +88,28 @@ public RepoConfig(final Vertx vertx, final Publisher<ByteBuffer> content) {
         );
     }
 
+    /**
+     * Repository URL.
+     *
+     * @return Async string of URL
+     */
+    public CompletionStage<URL> url() {
+        return this.repo().thenApply(
```

Comment:
@olegmoz  we can extract this part to a private method to avoid code duplication:
```java
this.repo().thenApply(
   map -> Objects.requireNonNull(map.string(${param}), ${message})
)
```